veggieseasons/lib/main.dart

```diff
@@ -16,16 +16,74 @@ void main() {
     DeviceOrientation.portraitDown,
   ]);
 
-  runApp(
-    ScopedModel<AppState>(
-      model: AppState(),
+  // The root scope is usually injected automatically by the
+  // CupertinoApp/MaterialApp. However, in this particular app there are widgets
+  // above the CupertinoApp that need their state restored, so we are injecting
+  // the root scope manually above everything.
+  runApp(RestorationScope.root(
+    restorationId: const RestorationId('root'),
+    child: VeggieApp(),
+  ));
+}
+
+// A new stateful widget that owns the [AppState], which contains state that
+// needs to be restored.
+class VeggieApp extends StatefulWidget {
+  const VeggieApp({Key key}) : super(key: key);
+
+  @override
+  State<VeggieApp> createState() => _VeggieAppState();
+}
+
+class _VeggieAppState extends State<VeggieApp> with RestorationMixin {
+  RestorableAppState appState = RestorableAppState();
+
+  @override
+  void initState() {
+    super.initState();
+    register(appState, const RestorationId('state'));
```

Comment:
could this name be more tightly scoped? `register` sounds really broad.

Reply:
I am open to suggestions :)

However, it was meant to have a somewhat broader scope because, after this call the RestorationMixin manages the registered proeprty.

Reply:
+1 to @xster's note. Something like `registerForRestoration` would help readers know immediately what the call is doing.

Reply:
Another idea I just had: `restoreProperty` because this is actually the call that retrieves the serialized value from the restoration state, deserializes it, and puts it into the property.